language-support/scala/codegen/src/main/scala/com/digitalasset/lf/codegen/lf/LFUtil.scala

```diff
@@ -293,37 +292,6 @@ final case class LFUtil(
                 $choiceMethod(${TermName(actorParamName)}, $dctorName(..$dargs))"""
       }.toList
   }
-
-  def templateCount(interface: EnvironmentInterface): Int = {
-    interface.typeDecls.count {
-      case (_, InterfaceType.Template(_, _)) => true
-      case _ => false
-    }
-  }
-
-  private[this] def foldTemplateReferencedTypeDeclRoots[Z](interface: EnvironmentInterface, z: Z)(
-      f: (Z, ScopedDataType.Name) => Z
-  ): Z =
-    interface.typeDecls.foldLeft(z) {
-      case (z, (id, InterfaceType.Template(_, tpl))) =>
-        tpl.foldMap(typ => Util.genTypeTopLevelDeclNames(typ).toSet).foldLeft(f(z, id))(f)
-      case (z, _) => z
-    }
-
-  protected[this] def precacheVariance(
-      interface: EnvironmentInterface
-  ): ScopedDataType.Name => ImmArraySeq[Variance] = {
-    import UsedTypeParams.ResolvedVariance
-    val resolved = foldTemplateReferencedTypeDeclRoots(interface, ResolvedVariance.Empty) {
-      (resolved, id) => resolved.allCovariantVars(id, interface)._1
-    }
-    id => resolved.allCovariantVars(id, interface)._2
-  }
-
-  private[this] lazy val precachedVariance = precacheVariance(iface)
-
-  def variance(sdt: ScopedDataType[_]): Seq[Variance] =
-    precachedVariance(sdt.name)
```

Comment:
Since this didn't have any dependency to anything else in the `LFUtil` package, I moved it out and made a class out of it (which was, by the way, used only in one place).

Reply:
A possible #13764 culprit; when this was done in util it happened at most once per datatype for the whole run; now any shared datatypes must have recomputed variance per referencing datatype.